xtask/src/run.rs

```diff
@@ -57,11 +57,12 @@ pub fn run(opts: Options) -> Result<(), anyhow::Error> {
     args.push(bin_path.as_str());
     args.append(&mut run_args);
 
-    // spawn the command
-    let err = Command::new(args.first().expect("No first argument"))
+    // run the command
+    let status = Command::new(args.first().expect("No first argument"))
         .args(args.iter().skip(1))
-        .exec();
+        .status()
+        .expect("failed to run the command");
 
-    // we shouldn't get here unless the command failed to spawn
-    Err(anyhow::Error::from(err).context(format!("Failed to run `{}`", args.join(" "))))
+    assert!(status.success());
```

Comment:
I think we should still fail with a descriptive message, not just with an assertion. So something like:

```suggestion
    if !status.success() {
        anyhow::bail!("Failed to run `{}`", args.join(" "));
    }
```

Reply:
then we probably should do the same with `build_ebpf.rs` as well = show the cargo command with arguments when it fails?